services/dashboard/routes/workers.go

```diff
@@ -0,0 +1,153 @@
+package routes
+
+import (
+	"encoding/base64"
+	"net/http"
+
+	"github.com/go-chi/chi/v5"
+
+	"sensorbucket.nl/sensorbucket/internal/web"
+	"sensorbucket.nl/sensorbucket/pkg/api"
+	"sensorbucket.nl/sensorbucket/services/dashboard/views"
+)
+
+type WorkerPageHandler struct {
+	router chi.Router
+	client *api.APIClient
+}
+
+func CreateWorkerPageHandler(client *api.APIClient) *WorkerPageHandler {
+	handler := &WorkerPageHandler{
+		router: chi.NewRouter(),
+		client: client,
+	}
+	handler.SetupRoutes(handler.router)
+	return handler
+}
+
+func (h WorkerPageHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	h.router.ServeHTTP(w, r)
+}
+
+func (h *WorkerPageHandler) SetupRoutes(r chi.Router) {
+	r.Get("/", h.listWorkers())
+	r.Get("/create", h.createWorkerPage())
+	r.Post("/create", h.createWorker())
+	r.Get("/{id}", h.workerDetails())
+	r.Patch("/{id}", h.updateWorker())
+}
+
+func (h *WorkerPageHandler) listWorkers() http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		page := &views.WorkerListPage{}
+		res, _, err := h.client.WorkersApi.ListWorkers(r.Context()).Execute()
+		if err != nil {
+			web.HTTPError(w, err)
+			return
+		}
+		page.Workers = res.Data
+
+		if isHX(r) {
+			page.WriteBody(w)
+			return
+		}
+		views.WriteIndex(w, page)
+	}
+}
+
+func (h *WorkerPageHandler) workerDetails() http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		page := &views.WorkerEditorPage{}
+		workerID := chi.URLParam(r, "id")
+		res, _, err := h.client.WorkersApi.GetWorker(r.Context(), workerID).Execute()
+		if err != nil {
+			web.HTTPError(w, err)
+			return
+		}
+		resUC, _, err := h.client.WorkersApi.GetWorkerUserCode(r.Context(), workerID).Execute()
+		if err != nil {
+			web.HTTPError(w, err)
+			return
+		}
+		page.Worker = res.Data
+		page.UserCode = resUC.GetData()
+
+		if isHX(r) {
+			page.WriteBody(w)
+			return
+		}
+		views.WriteIndex(w, page)
+	}
+}
+
+func (h *WorkerPageHandler) updateWorker() http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		workerID := chi.URLParam(r, "id")
+
+		var dto api.UpdateWorkerRequest
+		if err := r.ParseForm(); err != nil {
+			web.HTTPError(w, web.NewError(http.StatusBadRequest, "Bad request", ""))
+			return
+		}
+		if desc := r.FormValue("description"); desc != "" {
+			dto.Description = &desc
+		}
+		switch r.FormValue("state") {
+		case "on":
+			dto.SetState(2)
```

Comment:
should state not be an enum or some sort of constant?

Reply:
Yes, though I am not certain how to do that yet, as the enum is in the UserWorkers service package but the dashboard only utilizes the generated API.

I think the enums should atleast be converted to strings - such as "active" or "inactive" - to be more clear.

What do you think?